fms_extras/utils/generation.py

```diff
@@ -3,9 +3,623 @@
 from typing import Any, Callable, Dict, List, MutableMapping, Optional, Tuple, Union
 
 import torch
+import torch.nn as nn
 import torch.nn.functional as F
 
-from fms_extras.utils.cache.paged import PagedKVCacheManager
+from fms_extras.models.speculator import MLPSpeculator, apply_index_map, flatten_batch
+from fms_extras.utils.cache.paged import PagedAttentionCacheData, PagedKVCacheManager
+
+
+def __execute_prefill(
+    model: Union[nn.Module, Callable],
+    input_ids: Union[torch.Tensor, List[torch.Tensor]],
+    cache_data: PagedAttentionCacheData,
+    max_seq_len: int,
+) -> Tuple[torch.Tensor, torch.Tensor]:
+    """Execute the prefill and return information needed for first prediction
```

Comment:
I think we should explain what we mean by "prefill"

Reply:
fixed in latest commit